python/lsst/analysis/drp/scatterPlot.py

```diff
@@ -0,0 +1,532 @@
+from lsst.pipe.tasks.dataFrameActions import MagColumnNanoJansky, SingleColumnAction
+from lsst.skymap import BaseSkyMap
+
+import lsst.pipe.base as pipeBase
+import lsst.pex.config as pexConfig
+
+from . import dataSelectors as dataSelectors
+from .plotUtils import generateSummaryStats, parsePlotInfo, addPlotInfo
+
+
+class ScatterPlotWithTwoHistsTaskConnections(pipeBase.PipelineTaskConnections,
+                                             dimensions=("tract", "skymap"),
+                                             defaultTemplates={"inputCoaddName": "deep",
+                                                               "plotName": "deltaCoords"}):
+
+    catPlot = pipeBase.connectionTypes.Input(doc="The tract wide catalog to make plots from.",
+                                             storageClass="DataFrame",
+                                             name="objectTable_tract",
+                                             dimensions=("tract", "skymap"),
+                                             deferLoad=True)
+
+    skymap = pipeBase.connectionTypes.Input(doc="The skymap for the tract",
+                                            storageClass="SkyMap",
+                                            name=BaseSkyMap.SKYMAP_DATASET_TYPE_NAME,
+                                            dimensions=("skymap",))
+
+    scatterPlot = pipeBase.connectionTypes.Output(doc="A scatter plot with histograms for both axes.",
+                                                  storageClass="Plot",
+                                                  name="scatterTwoHistPlot_{plotName}",
+                                                  dimensions=("tract", "skymap"))
+
+
+class ScatterPlotWithTwoHistsTaskConfig(pipeBase.PipelineTaskConfig,
+                                        pipelineConnections=ScatterPlotWithTwoHistsTaskConnections):
+
+    axisActions = ConfigurableActionStructField(
+        doc="The actions to use to calculate the values used on each axis.",
+        default={"xAction": MagColumnNanoJansky, "yAction": SingleColumnAction,
+                 "magAction": MagColumnNanoJansky},
+    )
+
+    axisLabels = pexConfig.DictField(
+        doc="Name of the dataframe columns to plot, will be used as the axis labek: {'x':, 'y':, 'mag':}",
+        keytype=str,
+        itemtype=str
+    )
+
+    selectorActions = ConfigurableActionStructField(
+        doc="Which selectors to use to narrow down the data for QA plotting.",
+        default={"mainFlagSelector": dataSelectors.MainFlagSelector},
+    )
+
+    highSnStatisticSelectorActions = ConfigurableActionStructField(
+        doc="Selectors to use to decide which points to use for calculating statistics.",
+        default={"statSelector": dataSelectors.HighSnSelector},
+    )
+
+    lowSnStatisticSelectorActions = ConfigurableActionStructField(
+        doc="Selectors to use to decide which points to use for calculating statistics.",
```

Comment:
doc strings here and above should distinguish what the difference is between `highSnStatisticSelectorActions` and `lowSnStatisticSelectorActions`; currently their descriptions are identical.